table/block_based_table_builder.cc

```diff
@@ -852,9 +987,14 @@ Status BlockBasedTableBuilder::Finish() {
   }
 
   IndexBuilder::IndexBlocks index_blocks;
-  auto s = r->index_builder->Finish(&index_blocks);
-  if (!s.ok()) {
-    return s;
+  auto index_builder_status = r->index_builder->Finish(&index_blocks);
+  if (index_builder_status.IsIncomplete()) {
```

Comment:
I think moving partitioned index block logic here would make the code easier to read?
```
if (status.IsInComplete()) {
    while (status.IsInComplete()) {
        // write partitioned index
    }
} else {
    if (!status.ok()) {
        return status;
    }
    // write meta
    // write index
}
```